File: src/Services/Patients/Patients.Domain/Entities/Patient.cs
using Core.Domain.Aggregates;

namespace Patients.Domain.Entities;

public class Patient : BaseEntity<int>, IAggregateRoot
{
    public int TenantId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int? UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public DateOnly? BirthDate { get; set; }
    public string? CpfEncrypted { get; set; }
    public string Status { get; set; } = "active";
    public string TreatmentStatus { get; set; } = "screening";
    public string? EmergencyContactName { get; set; }
    public string? EmergencyContactPhone { get; set; }
    public DateTime? DeactivatedAt { get; set; }
    public string? DeactivationReason { get; set; }
}

File: src/Web/PsiFlow.Web/src/types/patient.ts
export type Patient = {
  id: number;
  tenantId?: number;
  name: string;
  fullName?: string;
  email: string;
  phone: string;
  birthDate?: string | null;
  status: 'Ativo' | 'Aguardando' | 'Inativo';
  treatmentStatus?: string;
  emergencyContactName?: string | null;
  emergencyContactPhone?: string | null;
  nextSession: string;
  risk: 'Baixo' | 'Moderado' | 'Alto';
};

export type PatientInvite = {
  patientId: number;
  email: string;
  phone: string;
};

export type PatientStatusUpdate = {
  treatmentStatus: string;
  reason?: string;
};
